packages/caravan-fees/src/tests/transactionAnalyzer.fixtures.ts

```diff
@@ -0,0 +1,125 @@
+      absoluteFee: "1192",
+      targetFeeRate: 10,
+      network: Network.MAINNET,
+      dustThreshold: 546,
+      changeOutputIndex: 3,
+    },
+    expected: {
+      txid: "d66730234997ea82005838c72e56470164fd6fb31c192791f898d9eadfad84a6",
+      canRBF: false,
+      canCPFP: true,
+      inputCount: 1,
+      outputCount: 8,
+      vsize: 330.5,
+      weight: 1322,
+      fee: "1192",
+      feeRate: "3.61",
+      recommendedStrategy: FeeBumpStrategy.CPFP,
+      estimatedRBFFee: 2118, // assuming full RBF is allowed
+      estimatedCPFPFee: 10732,
+      inputSequences: [4294967295],
+      outputValues: [
+        147872, 2803883, 151680, 389515, 699910, 780487, 38734, 17168362,
+      ],
+    },
+  },
+  {
+    // https://mempool.space/tx/78d249155c3d1a70353192a86d4770a2fbaa6ca2c07a0a6c7b0c3951c649d462
+    name: "Both RBF and CPFP possible",
+    test: {
+      txHex:
+        "020000000001013d8c8999e448c6593d7567534f36cd29d720931d2babc95fdd8074c7619f994d0200000000fdffffff0849a4000000000000160014e652c05df205783da3c52389297b735cbef809c7a3a8000000000000160014c8566e21fb0856d80619c5da034ac4feddc613d08abe000000000000160014539422c6b02c4f8ebd867ddefddc8fe1b826e80468bf000000000000160014dbe8fcf64788f7c0b3ccba91e164d5460648e7b190e20000000000001600148b90d42b9e07d2374325889126fa1ab0abd6e8a890e2000000000000160014f48f05c1c21cec9f26421bff7cb4e4ad0313fe19c54c020000000000160014ddd7bf2a9b1403fc7f6ab758057b3c6ad522af030d11b00000000000160014690b6f1113ddcad9eb2c931ad53b25cdfa756c24024730440220728d7f4df67f6f1fb0742bcd0a23b1604d561d8f9525ec85310a4f375a03100602204792910102db2001ed3bde8145e63dcae90e5aeef9bc16c21c4597baf3fba3ea012103f38818a20f1b81ee305b7e6dd9e0f76623a95c79ea17673a46dcfa558cddfe23a71a0d00",
+      absoluteFee: "2289",
+      targetFeeRate: 15,
+      network: Network.MAINNET,
+      dustThreshold: 546,
+      changeOutputIndex: 7,
+    },
+    expected: {
+      txid: "78d249155c3d1a70353192a86d4770a2fbaa6ca2c07a0a6c7b0c3951c649d462",
+      canRBF: true,
+      canCPFP: true,
+      inputCount: 1,
+      outputCount: 8,
+      vsize: 326.25,
+      weight: 1305,
+      fee: "2289",
+      feeRate: "7.02",
+      recommendedStrategy: FeeBumpStrategy.RBF,
+      estimatedRBFFee: 2617,
```

Comment:
So, I do have some concerns about these tests now (and I think this matches with my comment on this field in the code). Notice your test has the `absoluteFee` set to `2289` and here the `estimatedRBFFee` in the expectation is `2617`. This means that the analyzer is suggesting an RBF fee of only `328` more sats for rbf fee bumping. This would almost certainly fail BIP125 rule number 4:

> The replacement transaction must also pay for its own bandwidth at or above the rate set by the node's minimum relay fee setting. For example, if the minimum relay fee is 1 satoshi/byte and the replacement transaction is 500 bytes total, then the replacement must pay a fee at least 500 satoshis higher than the sum of the originals.

Specifically:

> the replacement must pay a fee at least 500 satoshis higher than the sum of the originals.

So if the original transaction is paying 600 sats then to pay 500 sats "higher than the sum of the original" would mean the new tx replacement tx should pay 1100 sats (or something like that).

(sum of the originals is to account for replacements of multiple txs, for example if you cancel or replace a tx that has children of its own, but we don't need to worry about that for now).

Luckily, mempool.space actually does show replaced txs, so you can look at a transaction that was replaced and what it's original fee and fee rate was and then the fee and fee rate that replaced it. Note [this](https://mempool.space/tx/f8fffbf4fd246b08143b8cbf93c14120ba779dd8be6e610f90feaddd427e20a3) for example. The fee started at 141 sats (1 sat/vbyte) then went to 242 sats (2 sat/vbyte) then to 423  sats (3 sats/vbyte). Notice how the total fees paid are nearly doubled each time (incidentally for almost the same size tx, so I'm guessing value was just taken out of the change).

Reply:
Ah ok! I stand corrected! I had to re-read rules 3 and 4 a couple more times but yeah I see I was wrong. 

Question: are there fixtures in bitcoin cores test suite that you could use to test against? It would be great if we could confirm our estimation against the reference implementation (with links included).

Reply:
> Question: are there fixtures in bitcoin cores test suite that you could use to test against? It would be great if we could confirm our estimation against the reference implementation (with links included).

[checked](https://github.com/bitcoin/bitcoin/pull/6871/files#diff-366a1e4f0784d362a06155c147c723d28e530fba3a121a839dd31438b562d84a) but I don't think they have fixtures that we can use ... :(